lib/app/features/nostr/providers/nostr_notifier.c.dart

```diff
@@ -33,12 +36,21 @@ class NostrNotifier extends _$NostrNotifier {
     ActionSource actionSource = const ActionSourceCurrentUser(),
     bool cache = true,
   }) async {
-    final relay = await _getRelay(actionSource);
-    await relay.sendEvents(events);
-    if (cache) {
-      return events.map(_parseAndCache).toList();
-    }
-    return null;
+    final dislikedRelaysUrls = <String>[];
+    NostrRelay? relay;
+    return withRetry(
+      () async {
+        relay = await _getRelay(actionSource, dislikedUrls: dislikedRelaysUrls);
+        await relay!.sendEvents(events);
+        if (cache) {
+          return events.map(_parseAndCache).toList();
+        }
+        return null;
+      },
+      onRetry: () {
+        dislikedRelaysUrls.add(relay!.url);
```

Comment:
`relay!` may throw in case the `withRetry`'s task fails inside `_getRelay`

Reply:
correct, fixed.